pkg/handlers/generic/mutation/imageregistries/credentials/inject.go

```diff
@@ -172,9 +200,8 @@ func (h *imageRegistriesPatchHandler) Mutate(
 				commands...,
 			)
 
-			generateErr = createSecretIfNeeded(ctx, h.client, registriesWithOptionalCredentials, clusterKey)
-			if generateErr != nil {
-				return generateErr
+			if err := client.ServerSideApply(ctx, h.client, credentialsSecret, client.ForceOwnership); err != nil {
+				return fmt.Errorf("failed to apply Image Registry Credentials Secret: %w", err)
```

Comment:
Why not do this when setting the owner ref above outside of the mutation?

Reply:
IIUC, in theory, both mutations may be skipped.

The handler should have side effects only if some mutation is executed.

Creating a resource is a side effect, so it must be done only if the mutation is executed.

Reply:
Ok could we make sure it's only applied a maximum of once then?

Reply:
I don't understand the motivation.

Do we want the two `patches.MutateIfApplicable` calls to be mutually exclusive?

If so, then we should make them mutually exclusive, and that would, in turn, guarantee that we apply the ownerRef at most once.

If we do not want the calls to be mutually exclusive, then I don't understand why we should apply the ownerRef at most once.

Reply:
Just an optimisation, reducing unnecessary calls to the API server.